experiment/compatibility-versions/kind-upgrade.sh

```diff
@@ -74,8 +74,6 @@ parse_args()
     done
 }
```

Comment:
TODO: all of this input computation should ideally come before any other logic, near the top of the script.

the various helper methods that aren't related to args/env inputs should come after